source/server/options_impl.cc

```diff
@@ -270,7 +270,7 @@ OptionsImpl::OptionsImpl(std::vector<std::string> args,
   drain_time_ = std::chrono::seconds(drain_time_s.getValue());
   parent_shutdown_time_ = std::chrono::seconds(parent_shutdown_time_s.getValue());
   socket_path_ = socket_path.getValue();
-  socket_mode_ = StringUtil::octalToDecimal(socket_mode.getValue());
+  StringUtil::atoull(socket_mode.getValue().c_str(), reinterpret_cast<uint64_t&>(socket_mode_), 8);
```

Comment:
This cast isn't safe; the callee could overwrite adjacent memory. Use a temporary uint64_t and then assign to `socket_mode_`.

Reply:
Done